rts/Lua/LuaTracyExtra.cpp

```diff
@@ -0,0 +1,90 @@
+/* This file is part of the Recoil engine (GPL v2 or later). */
+
+#include "LuaTracyExtra.h"
+
+#include "LuaInclude.h"
+#include "LuaUtils.h"
+
+#include "System/Misc/TracyDefs.h"
+#include <common/TracyQueue.hpp>
+
+#include <functional>
+#include <set>
+#include <string>
+
+/* Tracy seems to want unique, unchanging strings to be passed to
+ * its API, so we need to immanentize the ephemeral Lua strings
+ * by storing them.
+ *
+ * NB: strings here are never really cleaned up, but the use case assumes
+ * that they live a long time and there's just a handful of them. */
+static std::set <std::string, std::less<>> tracyLuaPlots;
+
+static const std::string& GetImmanentPlotName(const char *plotName)
+{
+	const auto plot = tracyLuaPlots.find(plotName);
+	if (plot != tracyLuaPlots.end())
+		return *plot;
+
+	return *tracyLuaPlots.emplace(plotName).first;
+}
+
+/*** Configure custom appearance for a Tracy plot for use in debugging or profiling
+ *
+ * @function tracy.LuaTracyPlotConfig
+ * @string plotName name of the plot to customize
+ * @string[opt="Number"] plotFormatType "Number"|"Percentage"|"Memory"
+ * @bool[opt=true] stepwise stepwise chart
+ * @bool[opt=false] fill whether to fill color
+ * @number[opt=white] color unit32 number as BGR color
```

Comment:
In theory "white" is not a valid Lua value, but then 0xFFFFFF isn't either, and all the real ways to actually produce that value from lua are too verbose and don't even convey the format well.

Reply:
but 0xFFFFFF is valid isn't it? or is the number you need something different? when I do Spring.Echo(0xFFFFFF) I get 16777215.

Reply:
also maybe that should read uint32 instead of unit32?

Reply:
Good catch, fixed

Reply:
anyways maybe the opt=white is fine, but doc at the end explaining how to set it would go a long way too, ie, like: "for example: 0xFFFFFF"